lib/puppet/provider/f5_sslcertificate/rest.rb

```diff
@@ -0,0 +1,41 @@
+require 'puppet/provider/f5'
+require 'json'
+
+Puppet::Type.type(:f5_sslcertificate).provide(:rest, parent: Puppet::Provider::F5) do
+
+  def create_message(basename, hash)
+    # Create the message by stripping :present.
+    new_hash            = hash.reject { |k, _| [:ensure,:name, :provider, Puppet::Type.metaparams].flatten.include?(k) }
+  #  new_hash[:name]     = basename
+
+    return new_hash
+  end
+
+  def message(object)
+    # Allows us to pass in resources and get all the attributes out
+    # in the form of a hash.
+    message = object.to_hash
```

Comment:
This seems to be used in a lot of providers.  I've no idea why though.  `object` is already a hash, isn't it?

Reply:
@ericzji offline please confirm value of object passed in is already hash.  Nice find @alexjfisher.

Reply:
the class for object  is Puppet::Type::F5_sslcertificate. Not very clear the structure, but different from hash? @alexjfisher @abrader